storm-core/src/jvm/backtype/storm/scheduler/resource/RAS_Node.java

```diff
@@ -230,6 +241,61 @@ public void freeTopology(String topId, Cluster cluster) {
         _topIdToUsedSlots.remove(topId);
     }
 
+    private void freeMemory(double amount) {
+        _availMemory += amount;
+        LOG.debug("freeing {} memory on node {}...avail mem: {}", amount, this.getHostname(), _availMemory);
+        if (_availMemory > this.getTotalMemoryResources()) {
+            LOG.warn("Freeing more memory than there exists!");
```

Comment:
1. Grammar. 
2. Should this be a simple warn? Sounds like a more serious condition, maybe throw an exception?